### ISSUE TITLE
to_latex with MI column and index names

### ISSUE THREAD

**Issue by hayd** [pandas-dev/pandas#8336]
I introduced/uncovered a bug here with #8219.

If you pass in DataFrame with MI index and col names it doesn't output correctly (and never did). Should be able to fix pretty easily, since it's my fault am assigning myself!

**Comment by jorisvandenbossche** [comment]
I also just stumbled upon this. And it does seem to be a regression, as it worked in 0.14:

```
In [1]: idx = pd.MultiIndex.from_product([['A0', 'A1', 'A2'],['B0','B1']])

In [2]: df = pd.DataFrame(np.random.random((6, 2)), index=idx, columns=['col1','
col2'])

In [3]: print df.to_latex()
\begin{tabular}{lrr}
\toprule
{} &      col1 &      col2 \\
A0 B0 &  0.465015 &  0.706859 \\
\midrule
   B1 &  0.768717 &  0.528846 \\
A1 B0 &  0.762849 &  0.801075 \\
   B1 &  0.391274 &  0.751734 \\
A2 B0 &  0.647738 &  0.039428 \\
   B1 &  0.437429 &  0.512462 \\
\bottomrule
\end{tabular}
```

and now in 0.16.0 the same gives:

```
In [53]: print df.to_latex()
\begin{tabular}{llrr}
\toprule
   &    &      col1 &      col2 \\
\midrule
A0 & B0 &  0.512408 &  0.441161 \\
A1 & B1 &  0.456337 &  0.472716 \\
\bottomrule
\end{tabular}
```

**Comment by toobaz** [comment]
I think this was fixed:

```
In [3]: idx = pd.MultiIndex.from_product([['A0', 'A1', 'A2'],['B0','B1']])

In [4]: df = pd.DataFrame(np.random.random((6, 2)), index=idx, columns=['col1','col2'])

In [5]: print(df.to_latex())
\begin{tabular}{llrr}
\toprule
   &    &      col1 &      col2 \\
\midrule
A0 & B0 &  0.184294 &  0.850768 \\
   & B1 &  0.762865 &  0.842426 \\
A1 & B0 &  0.137573 &  0.266622 \\
   & B1 &  0.365801 &  0.512295 \\
A2 & B0 &  0.449834 &  0.631251 \\
   & B1 &  0.374689 &  0.269184 \\
\bottomrule
\end{tabular}

```

**Comment by jorisvandenbossche** [comment]
Yes, this was actually already fixed in 0.16.1: closed by https://github.com/pandas-dev/pandas/pull/9908
